tests/test_flow/test_persistence_random.py

```diff
@@ -194,21 +259,77 @@ class ModelBinding:
     dep_entity_names = attr.ib()
     use_tuples_for_output = attr.ib()
     func_version = attr.ib(default=1)
+    annotated_func_version = attr.ib(default=1)
 
     has_persisted_values = attr.ib(default=False)
     persisted_values_by_entity_name = attr.ib(default=None)
+    persisted_values_stale_ancestor = attr.ib(default=None)
+
     some_descendant_might_have_persisted_values = attr.ib(default=False)
+    some_descendant_might_have_fresh_persisted_values = attr.ib(default=False)
+
+    def update_func_version(self, change_func, update_annotation, versioning_mode):
```

Comment:
nit: This name is a little confusing now that don't always change the func or the annotation. Can we call this `update` or `update_binding`?